### PR TITLE
DBEX: format zip code to meet EMMS specifications

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

The processing of 4142 forms, as part of the 526 form submission workflow, has experienced issues where silent failures were triggered as a result of Central Mail EMMS API regex validations, including:
- the presence of special characters in a Veteran's first and/or last names
- an invalid zip code format

After some clarification concerning actual expectations, the following modifications have been made to the generated metadata used to populate the request body of a 4142 form submitted to CentralMail for processing:
- the removal of any apostrophes present in a veteran's first and/or last name
- the normalization of postal codes to conform to a 5-digit or hyphenated 9-digit format

## Related issue(s)

Zenhub ticket [#72321](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/72321)

## Testing done

Test added to validate zip code reformatting meets expectations, and modification of the test that characters in a veteran's first/last name match the allowed regex.

## What areas of the site does it impact?

Form 4142 processing as part of Form 526 async submission

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by tblackwe — DISMISSED** [review]

**Review by mchae-nava — APPROVED** [review]
LGTM

**Review by sethdarragile6 — APPROVED** [review]

File: spec/sidekiq/central_mail/submit_form4142_job_spec.rb [hunk 1]
```diff
@@ -28,6 +28,16 @@
                                form_json:,
                                submitted_claim_id: evss_claim_id)
     end
+    let(:metadata_hash) do
+      form4142 = submission.form[Form526Submission::FORM_4142]
+      form4142['veteranFullName'].update('first' => 'Beyoncé', 'last' => 'Knowle$-Carter')
```
Comment (rmtolmach) [inline spec/sidekiq/central_mail/submit_form4142_job_spec.rb:33]: From your description:
> the removal of any apostrophes present in a veteran's first and/or last name

I don't think Beyoncé would test that scenario. I was thinking an apostrophe in a name would be something like `O'Brian`. Can you include a test for that?

Comment (freeheeling) [reply spec/sidekiq/central_mail/submit_form4142_job_spec.rb]: Included an apostrophe in the set up. Thanks.

**Comment by freeheeling** [comment]
Closing to instead integrate with pending [MetadataValidator update](https://github.com/department-of-veterans-affairs/vets-api/pull/15030).

**Pull request closed without merging**